src/gui/editors/SongEditor.cpp

```diff
@@ -291,42 +297,40 @@ void SongEditor::loadSettings( const QDomElement& element )
 
 
 
+/*! \brief Return grid size as number of bars */
 float SongEditor::getSnapSize() const
 {
-	// 1 Bar is the third value in the snapping dropdown
-	int val = -m_snappingModel->value() + 3;
+	float snapSize = SNAP_SIZES[m_snappingModel->value()];
+
 	// If proportional snap is on, we snap to finer values when zoomed in
 	if (m_proportionalSnap)
 	{
-		val = val - m_zoomingModel->value() + 3;
+		// Finds the closest available snap size
+		const float optimalSize = snapSize * DEFAULT_PIXELS_PER_BAR / pixelsPerBar();
+		return *std::min_element(SNAP_SIZES.begin(), SNAP_SIZES.end(), [&](float a, float b)
```

Comment:
```suggestion
		return *std::min_element(SNAP_SIZES.begin(), SNAP_SIZES.end(), [optimalSize](float a, float b)
```
Can capture by value instead of reference